dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/VisitorOpsComposite.java

```diff
@@ -95,9 +90,9 @@ public Void visitOpVectorShuffle(SpirvParser.OpVectorShuffleContext ctx) {
         List<Expression> concat = new ArrayList<>();
         for (Integer index : components) {
             if (index >= 0 && index < s1) {
-                concat.add(v1.getOperands().get(index));
+                concat.add(v1 instanceof ConstructExpr ? v1.getOperands().get(index) : expressions.makeExtract(v1, index));
             } else if (index >= s1 && index < s1 + s2) {
-                concat.add(v2.getOperands().get(index - s1));
+                concat.add(v2 instanceof ConstructExpr ? v2.getOperands().get(index - s1) : expressions.makeExtract(v2, index - s1));
```

Comment:
Why do you need to do the case distinction? Don't you trust the expression simplifier? :)

Reply:
This has nothing to do with the expression simplifier. It was rather causing problems already at parsing time (the one I told you yesterday night).

My previous understanding of `InsertExpr.getOperand()` was wrong. Calling `insert([old_0,...], new_0).getOperands()` returns `[old_0,...,new_0]` and not `[new_0,...]`.

Reply:
I understand that the previous code was wrong, but why don't you just do the extract at all times?
Now, both cases of the ternary operator are semantically identical, the former is just a step the simplifier would otherwise do.